Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/hibernate/AuditRepositoryDAO.java

```diff
@@ -153,12 +157,178 @@ public List queryAuditRepositoryOnCriteria(String eUserId, String ePatientId, Da
+            LOG.info("Getting Record for Audit Viewer ");
+
+            // Build the criteria
+            Criteria queryCriteria = session.createCriteria(AuditRepositoryRecord.class);
+
+            if (NullChecker.isNotNullish(messageId)) {
+                if (messageId.startsWith(NhincConstants.WS_SOAP_HEADER_MESSAGE_ID_PREFIX)) {
+                    queryCriteria.add(Restrictions.eq("messageId", messageId));
+                } else {
+                    queryCriteria.add(Restrictions.eq("messageId",
+                        NhincConstants.WS_SOAP_HEADER_MESSAGE_ID_PREFIX + messageId));
+                }
+            }
+
+            if (NullChecker.isNotNullish(relatesTo)) {
+                queryCriteria.add(Restrictions.eq("relatesTo", relatesTo));
             }
+            // if no criteria is passed then it will search full database with above mentioned columns in the result
+            queryList = setProjectionFields(queryCriteria).list();
+        } catch (HibernateException e) {
+            LOG.error("Exception in AuditLog.get() occurred due to :" + e.getLocalizedMessage(), e);
+        } finally {
+            closeSession(session, false);
         }
         return queryList;
     }
+
+    /**
+     * This method does a query to database to get the Audit Log Messages based different options
+     *
+     * @param outcome
+     * @param startDate
+     * @param userId
+     * @param eventTypeList
+     * @param remoteHcidList
+     * @param endDate
+     * @return List
+     */
+    public List queryByAuditOptions(Integer outcome, List<String> eventTypeList, String userId,
+        List<String> remoteHcidList, Date startDate, Date endDate) {
+
+        Session session = null;
+        List<AuditRepositoryRecord> queryList = null;
+        try {
+            SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+            session = sessionFactory.openSession();
+
+            LOG.info("Getting Record for Audit Query ");
+
+            // Build the criteria
+            Criteria queryCriteria = session.createCriteria(AuditRepositoryRecord.class);
+
+            if (outcome != null && outcome >= 0) {
+                queryCriteria.add(Restrictions.eq("outcome", outcome));
+            }
+
+            if (NullChecker.isNotNullish(eventTypeList)) {
+                queryCriteria.add(Restrictions.in("eventType", eventTypeList));
+            }
+
+            if (NullChecker.isNotNullish(userId)) {
+                queryCriteria.add(Restrictions.eq("userId", userId));
+            }
+
+            if (NullChecker.isNotNullish(remoteHcidList)) {
+                queryCriteria.add(Restrictions.in("remoteHcid", remoteHcidList));
+            }
+
+            if (startDate != null && endDate != null) {
+                queryCriteria.add(Expression.between("eventTimestamp", new Date(startDate.getTime()),
+                    new Date(endDate.getTime())));
+            } else if (startDate != null && endDate == null) {
```

Comment:
Do we need to create a new Date object here? startDate and endDate are already of type Date